src/js/apps/patients/worklist/worklist_app.js

```diff
@@ -128,17 +165,40 @@ export default App.extend({
     }
 
     this.worklistId = worklistId;
-    this.initListState();
 
     this.showView(new LayoutView({
       worklistId: this.worklistId,
     }));
 
+    this.initListState();
     this.getRegion('list').startPreloader();
     this.showSortDroplist();
     this.showTableHeaders();
     this.showTooltip();
+  },
+  beforeStart() {
+    const filter = this.getState().getEntityFilter();
+    return Radio.request('entities', `fetch:${ this.getState().getType() }:collection`, { filter });
+  },
+  onStart(options, collection) {
+    this.collection = collection;
 
+    this.collectionView = new ListView({
+      collection: this.collection,
+      state: this.getState(),
+      viewComparator: this.getComparator(),
+    });
+
+    this.showChildView('list', this.collectionView);
+
+    if (_.keys(this.getState().getSelected(this.collection)).length) {
+      this.showMultiEditButtonView();
+      return;
+    }
+
+    this.startFiltersApp();
+  },
+  startFiltersApp() {
```

Comment:
hmm moving this to onStart is unfortunate.  it means that the filters will be reset every time they're changed..  So .. you pick a value in the filter.. and then realize you picked the wrong one.. so you start to pick the other one.. but during that the first request comes in and re-starts the filters and your droplist is now replaced with a new one.. 

I think what we can do is just be ok with the filters showing up during the `onBeforeStart` and the multi-select showing `onStart`  and in `onStart` it'll be ok if `startFiltersApp` is called again since it'll immediately bail if it is already running.